src/progress.rs

```diff
@@ -257,13 +430,28 @@ impl ProgressSet {
     pub fn maximal_committed_index(&self) -> u64 {
```

Comment:
To make the API work for "membership change can work on received instead of applied", we need to add an argument `limit` (it's just Raft::pending_conf_index) to frobiden `BeginMembershipChange` to be committed by only old configuration.

Reply:
After think more about it, it's not necessary in fact, because when we fetch `Ready` from `RawNode` on leader, the `BeginMembershipChange` hasn't been broadcasted to other peers. So it will never be committed.
So please ignore this. :)

Reply:
Oh, amazing. Great thoughts. You are so smart. :) I'm jealous.